### ISSUE TITLE
Accessibility enhancements

### ISSUE THREAD

**Issue by smashism** [macadmins/nudge#307]
Hey y'all, I wanted to put in this request to see if two minor adjustments could be made to improve accessibility of the text in the Nudge prompt window.

1. "Important Notes" text. It's currently red which can cause challenges for readers that are colorblind. It may take some experimentation to find the right shade that works against the background for both light mode and dark mode (I'd imagine blue for light mode and orange for dark mode would be good options that would maintain a high enough contrast to be more easily readable). <br />
Apple has some documentation on color for visual design and accessibility here:
https://developer.apple.com/design/human-interface-guidelines/macos/visual-design/color/
https://developer.apple.com/design/human-interface-guidelines/accessibility/overview/color-and-contrast/<br />
There are some other handy resources out there too, like:
https://uxdesign.cc/color-blindness-in-user-interfaces-66c27331b858<br />
and Xcode has its own built-in accessibility inspector:
https://zonneveld.dev/accessibility-inspector/<br />

2. Improving contrast ratio in sidebar text. The integer column (where the required/current OS version, days, deferred counts reside) doesn't meet the standard 4.5 to 1 contrast ratio recommendation for accessibility. Should be a relatively simple color tweak to improve the ratio (something the Xcode accessibility checker can check).

There are other things that the accessibility checker may flag (e.g., spacing, element layouts) but I think focusing on font colors and contrast is a great place to start.

Happy to talk through this further with anyone that has the cycles to look into this. 💙

**Comment by erikng** [comment]
As a red/green colorblind person, the red one is interesting to me. I've never seen an issue with it and I'm _very_ sensitive to it since red is one of my worst colors that I can see.

As for contrast ratio, could you potentially show in a screenshot what exactly you mean?

**Comment by colorenz** [comment]
Maybe it is possible that we get a preference key to determine the color? So that you can enter a hex value?

 // mainContentNote
                    HStack {
                        Text(mainContentNote)
                            .font(.callout)
                            .fontWeight(.bold)
                            **.foregroundColor(Color.red)**
                        Spacer()
                    }

**Comment by erikng** [comment]
@bartreardon would know :)

**Comment by bartreardon** [comment]
yeah, that's easy to do - If you want you can specify hex value or one of the built in colours. I have code that can do that. happy to do a PR.

**Comment by bartreardon** [comment]
something like that?

**Comment by bartreardon** [comment]
So it occurs to me that specifying hex will be literal. When using the colour names, although limited they should adjust according to colour accessibility https://developer.apple.com/design/human-interface-guidelines/macos/visual-design/color/

**Comment by erikng** [comment]
So I have some thoughts.

I believe I have address point 1 with the following:

https://www.hackingwithswift.com/books/ios-swiftui/supporting-specific-accessibility-needs-with-swiftui
https://davidmathlogic.com/colorblind/#%23D81B60-%231E88E5-%23FFC107-%23004D40

By setting the following in accessibility
<img width="780" alt="Screen Shot 2022-02-22 at 10 59 43 AM" src="https://user-images.githubusercontent.com/5685016/155181937-08ce3c4c-89bd-4815-89eb-d6cf90d293fe.png">

it will look like this
<img width="1012" alt="Screen Shot 2022-02-22 at 10 59 40 AM" src="https://user-images.githubusercontent.com/5685016/155181985-fe13d4ee-6b62-4f5f-9d72-329f236d482f.png">
<img width="1012" alt="Screen Shot 2022-02-22 at 11 01 43 AM" src="https://user-images.githubusercontent.com/5685016/155182098-23c0acef-d1d8-4b6b-a26a-fd4055a7e9b2.png">

vs this

<img width="1012" alt="Screen Shot 2022-02-22 at 10 59 33 AM" src="https://user-images.githubusercontent.com/5685016/155182065-a0b68a1b-9a0e-4ff4-9bb4-aa5d04633575.png">
<img width="1012" alt="Screen Shot 2022-02-22 at 11 00 52 AM" src="https://user-images.githubusercontent.com/5685016/155182127-caa8c841-b29d-4893-bedf-a46ea5e53b3f.png">

For point 2, again by setting the following in accessibility
<img width="780" alt="Screen Shot 2022-02-22 at 11 02 17 AM" src="https://user-images.githubusercontent.com/5685016/155182391-faf96b7b-c760-454b-a33b-d24bf7ddd18f.png">

it will look like this
<img width="1012" alt="Screen Shot 2022-02-22 at 11 02 10 AM" src="https://user-images.githubusercontent.com/5685016/155182438-68ec4e06-f99c-4d56-aaac-228faf2e8082.png">
<img width="1012" alt="Screen Shot 2022-02-22 at 11 01 58 AM" src="https://user-images.githubusercontent.com/5685016/155182441-9562e11a-3ca1-4b1c-be20-78da6c81a29d.png">

**Comment by erikng** [comment]
I'm going to make an executive decision if I don't hear back soon. I'm still waiting 8 days from my initial question and I personally believe these accessibility enhancements in https://github.com/macadmins/nudge/pull/323 will address the two concerns on this issue.

If I don't hear back by tomorrow, I'm going to merge this and close it.

**Comment by smashism** [comment]
I think the emphasize without color option that makes the font slightly bigger helps for sure. On the issue of contrast it has more to do with the degree of contracts between the font color and background color on the text itself. I'm struggling to get Xcode to compile the app locally so I can show you what I'm talking about, but if you go to View > Inspectors > Accessibility you should see results around what I'm talking about.

**Comment by erikng** [comment]
I'm struggling to understand here. The background color is the default swiftui background color and so is the font color. This implies that all swiftui applications would inherently have this issue of contrast.

If you look at the accessibility inspector itself, its font color RGB 37, 37, 37, which is identical to nudge's font color. The background color of accessibility showing a window is RGB 242, 242, 242 and Nudge is barely different (not sure why) if 243, 239, 243.

<img width="926" alt="Screen Shot 2022-02-24 at 9 22 22 AM" src="https://user-images.githubusercontent.com/5685016/155554035-9b8c7055-4685-4556-819c-91101f5f665e.png">

I'm not sure what I can do here except change the color itself and that likely presents more problems should apple in the future change the primary/secondary colors built into the OS.

**Comment by erikng** [comment]
I [read up on this](https://www.raywenderlich.com/7180582-ios-accessibility-in-swiftui-tutorial-part-2-organizing) for ios and learned how to better use this tool.
From what I can tell there's only one element that doesn't match the 4.5 min ratio and that's the `More Info` button.

<img width="779" alt="Screen Shot 2022-02-24 at 9 40 10 AM" src="https://user-images.githubusercontent.com/5685016/155556908-d33cb16e-5a3f-4c94-b050-9fa5039bb845.png">
<img width="779" alt="Screen Shot 2022-02-24 at 9 39 49 AM" src="https://user-images.githubusercontent.com/5685016/155556910-93e318bc-3ff6-49b0-a2d1-75b20657d0cd.png">
<img width="779" alt="Screen Shot 2022-02-24 at 9 39 02 AM" src="https://user-images.githubusercontent.com/5685016/155556911-e5ea25f2-f822-493e-a67c-38d04ff128b4.png">
<img width="779" alt="Screen Shot 2022-02-24 at 9 35 33 AM" src="https://user-images.githubusercontent.com/5685016/155556914-2a15cda0-f59b-47df-abd6-d12ff4910bde.png">

**Comment by erikng** [comment]
Actually I think I finally understand.

> The integer column 

FFS. Would help if I fully comprehended what you were saying in the beginning :)
